lib/gst/player/gstplayer.c

```diff
@@ -688,6 +688,13 @@ eos_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
   self->priv->is_eos = TRUE;
 }
 
+void gst_player_set_playback_rate (GstPlayer * self, gdouble rate)
+{
+  g_return_if_fail (GST_IS_PLAYER (self));
+  self->priv->rate = rate;
+  gst_player_seek (self, 0);
```

Comment:
Yes, see my comments. This all has to be done differently and refactored to have unified seeking logic for play speed and seeking.